lib/spack/spack/test/stage.py

```diff
@@ -728,6 +736,52 @@ def test_first_accessible_path(self, tmpdir):
         # Cleanup
         shutil.rmtree(str(name))
 
+    def test_create_stage_root(self, tmpdir, no_path_access):
+        """Test _create_stage_root permissions."""
+        spack_dir = tmpdir.join('path')
+
+        # Ensure an OS Error is raised on the created, non-user directory
+        with pytest.raises(OSError) as exc_info:
+            spack.stage._create_stage_root(str(spack_dir))
+
+        assert exc_info.value.args[0] == errno.EACCES
```

Comment:
That's what I thought when I tried to use it but I was getting errors so switched to using the `value`.

Reply:
Could you mention the errors? If there's some way to refer to `.errno` I'd prefer that. If there's no way around it (e.g. it isn't available for Python 3.5) then IMO a comment would be worthwhile.

Reply:
One problem is `exc_info` is an `ExceptionInfo` instance, not an `OSError` instance:

```E       AttributeError: 'ExceptionInfo' object has no attribute 'errno'```

I can use `exc_info.type.errno` and get:

```E       AssertionError: assert <member 'errno' of 'exceptions.EnvironmentError' objects> == 13
E        +  where <member 'errno' of 'exceptions.EnvironmentError' objects> = <type 'exceptions.OSError'>.errno
E        +    where <type 'exceptions.OSError'> = <ExceptionInfo OSError tblen=2>.type
E        +  and   13 = errno.EACCES
```

Would you prefer a `try-except` block where I explicitly catch the error instead of using `with pytest.raises`?

Reply:
> I can use exc_info.type.errno and get:

if you mean that you can replace 

```
assert exc_info.value.args[0] == errno.EACCES
```

with 

```
 assert exc_info.type.errno == errno.EACCES
```

then I would prefer that.

Reply:
> > I can use exc_info.type.errno and get:
> 
> if you mean that you can replace
> 
> ```
> assert exc_info.value.args[0] == errno.EACCES
> ```
> 
> with
> 
> ```
>  assert exc_info.type.errno == errno.EACCES
> ```
> 
> then I would prefer that.

I should've said "If i use `exc_info.type.errno` as in if I use it I get a different error.

Would you prefer I use a `try-except` block so `err.errno` is available?

Reply:
Thanks for the clarification - a `try-except` would I think be preferable in this case. From reading

https://docs.pytest.org/en/3.0.1/builtin.html

I think `assert exc_info.value.errno == errno.EACCES` might work too (i.e. replace `type` with `value`)

Reply:
I guess missed that substitution.  It does seem to work for 2.7.16 and 3.6.  Thanks.